### PR TITLE
chore: add `generateMappingWithGroupingKey(bytes6,bytes4,bytes20)` to LSP2Utils

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?

## ♻️ Refactor

- Add `generateMappingWithGroupingKey(bytes6,bytes4,bytes20)` to LSP2Utils
- Add foundry test to `generateMappingWithGroupingKey(bytes6,bytes4,bytes20)`

### PR Checklist

- [X] Wrote Tests
- [ ] Wrote Documentation
- [X] Ran `npm run linter`
- [x] Ran `npm run prettier`
- [X] Ran `npm run build`
- [X] Ran `npm run test`

### CODE REVIEW

**Review by skimaharvey — APPROVED** [review]
nice

**Review by b00ste — APPROVED** [review]

File: contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol [hunk 1]
```diff
@@ -144,6 +144,22 @@ library LSP2Utils {
         return bytes32(temporaryBytes);
     }
 
+    /**
+     * @dev Generate a data key of keyType MappingWithGrouping
+     * <bytes6keyPrefix>:<bytes4mapPrefix>:<bytes2(0)>:<subMapKey>
+     * @param keyPrefix Used for the first part (bytes6) of the data key of keyType MappingWithGrouping
```
Comment (CJ42) [inline contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol:150]: I think we can shorten by removing "Used for" and remove the type between parentheses, as the parameters already show them.

```suggestion
     * @param keyPrefix the first part of the data key of keyType MappingWithGrouping
```

File: tests/foundry/LSP2ERC725YJSONSchema/LSP2Utils.t.sol [hunk 2]
```diff
@@ -117,6 +119,23 @@ contract LSP2UtilsTests is Test {
         LSP2Utils.isCompactBytesArray(data);
     }
 
+    function testShouldConcatenateMappingWithGroupingKey(
```
Comment (CJ42) [inline tests/foundry/LSP2ERC725YJSONSchema/LSP2Utils.t.sol:122]: nice foundry tests 👌

**Review by CJ42 — APPROVED** [review]
added one review comment.

I think we should consider renaming the function names in `LSP2Utils` with `DataKeys` not`Key`.
To be discussed in the future.

**Pull request merged**